include/triton/common/logging.h

```diff
@@ -73,9 +73,30 @@ class Logger {
   // Get the logging format.
   Format LogFormat() { return format_; }
 
+  // Get the logging format as a string.
+  std::string LogFormatString()
+  {
+    std::string format = "default";
+    if (format_ == Format::kISO8601) {
+      format = "ISO8601";
+    }
+    return format;
+  }
+
   // Set the logging format.
   void SetLogFormat(Format format) { format_ = format; }
 
+  // Get the log output file name.
+  const std::string& LogFile() { return filename_; }
+
+  // Set the log output file.
+  void SetLogFile(const std::string& filename)
+  {
+    const std::lock_guard<std::mutex> lock(mutex_);
+    filename_ = filename;
+    file_name_changed_ = true;
+  }
```

Comment:
Why not just open the file stream here?

Reply:
That's a great idea! It has been incorporated in the latest commit.